plugins/device/client.go

```diff
@@ -0,0 +1,84 @@
+package device
+
+import (
+	"context"
+	"io"
+
+	"github.com/hashicorp/nomad/plugins/base"
+	"github.com/hashicorp/nomad/plugins/device/proto"
+	netctx "golang.org/x/net/context"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// devicePluginClient implements the client side of a remote device plugin, using
+// gRPC to communicate to the remote plugin.
+type devicePluginClient struct {
+	// basePluginClient is embedded to give access to the base plugin methods.
+	*base.BasePluginClient
+
+	client proto.DevicePluginClient
+}
+
+func (d *devicePluginClient) Fingerprint(ctx context.Context) (<-chan *FingerprintResponse, error) {
+	var req proto.FingerprintRequest
+	stream, err := d.client.Fingerprint(ctx, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	out := make(chan *FingerprintResponse, 1)
+	go d.handleFingerprint(ctx, stream, out)
+	return out, nil
+}
+
+// handleFingerprint should be launched in a goroutine and handles converting
+// the gRPC stream to a channel. Exits either when context is cancelled or the
+// stream has an error.
```

Comment:
Since callers of Fingerprint will need to know the semantics of the FingerPrintResponse chan I think we should move at least some of this comment up there and make the close vs error vs context.Canceled behavior explicit.